### PR TITLE
#5633 Add connector development guidelines

### PR DESCRIPTION
**Pull request by squakez**
Adding a full backend development guideline to help during onboarding or new contribution as specified in #5633

### CODE REVIEW

File: content/docs/connectors.md [hunk 1]
```diff
@@ -0,0 +1,634 @@
...
+menu:
+  sidenav:
+    name: Connectors development
+    weight: 3
+    parent: Developer Docs
+toc: true
+weight: 20
+---
+
+One of the most important features of Syndesis is to be able to connect to a variety of sources and destinations. The middleware developed is going to take care of transparently of the integration to such source/destination of data requiring a custom development of the specific transport involved in the chosen component.
```
Comment (Delawen) [inline content/docs/connectors.md]: This phrase looks a bit complex for me. I would simplify it, something like:

"The developed middleware will take care transparently of the integration of such data source/destination. This requires a custom development of the specific transport involved in the chosen component."

File: content/docs/connectors.md [hunk 2]
```diff
@@ -0,0 +1,634 @@
...
+
+Since Apache Camel is providing a large [offering of integrations components](https://camel.apache.org/components.html), Syndesis does offer the possibility to use them thanks to a proxy component whose goal is to identify the component and provide the correct configuration settings to make them available through the user interface.
+
+### Tutorial
+We will use as a reference the development of a connector for MongoDB and we will try to show explicitly all the configuration and development you will have to perform in order to develop your own connector.
+
+If you haven’t done yet, fork the syndesis project, check it out locally and familiarize with its structure. Also you may want to read the [engineering guidelines](https://syndesis.io/community/contributing/#1-fork-the-repository) before starting to work on it.
```
Comment (Delawen) [inline content/docs/connectors.md]: "haven't done *it* yet", right?

And I would use less commas and more full stops (points). But that's just my way of writing.

File: content/docs/connectors.md [hunk 3]
```diff
@@ -0,0 +1,634 @@
...
+### Camel upstream component
+The first thing you will have to do is to have a look at the list of camel components provided in the introduction. Fortunately there is quite a lot of components available, so you should be lucky as in the example I’m going to show for [MongoDB](https://camel.apache.org/mongodb.html).
```
Comment (Delawen) [inline content/docs/connectors.md]: "you should be lucky" or "you can be lucky"?

File: content/docs/connectors.md [hunk 4]
```diff
@@ -0,0 +1,634 @@
...
+###### Changes in existing camel components
+If you are going to make any changes to the existing camel components, in order to be able to use them properly make sure to bring the correct version and to rebuild the `camel-catalog` dependency. This one is used at runtime by Syndesis to read configuration information such as the expected parameters, so, any change must be reflected by refreshing this dependency as well.
+### Maven project structure
+As there is not yet an archetype available, we need to make some manual `pom.xml` configuration in order to prepare our project. In order to maintain the original structure of the project you will have to create a connector folder under the `app/connector` directory. In order to maintain the convention, call it `connector-nameOfComponent` (ie, `connector-mongodb`).
+
+Next step is to create the minimum viable pom in order to be able to wrap all the needed dependencies for our connector.
+###### Connector description
+Almost all related to the project is managed by the connector parent pom. So, you just have to provide the name of this new dependency:
```
Comment (Delawen) [inline content/docs/connectors.md]: "Almost *everything* related to"

File: content/docs/connectors.md [hunk 5]
```diff
@@ -0,0 +1,634 @@
...
+            <version>1.7-SNAPSHOT</version>
+            <relativePath>../pom.xml</relativePath>
+        </parent>
+    
+        <modelVersion>4.0.0</modelVersion>
+        <artifactId>connector-mongodb</artifactId>
+        <name>Connector :: MongoDB</name>
+        <packaging>jar</packaging>
+
+You will have to change the version and name of your connector accordingly.
```
Comment (Delawen) [inline content/docs/connectors.md]: Should we mention where can we find the proper version or would that be too dummy?

File: content/docs/connectors.md [hunk 6]
```diff
@@ -0,0 +1,634 @@
...
+
+There is a class you have to extend in order to benefit all the environment provided by Syndesis and Camel and just worry about making your testsuite: [io.syndesis.connector.support.test.ConnectorTestSupport](https://github.com/syndesisio/syndesis/blob/master/app/connector/support/test/src/main/java/io/syndesis/connector/support/test/ConnectorTestSupport.java) that is extending the upstream Camel supporting test class [org.apache.camel.test.junit4.CamelTestSupport](https://github.com/apache/camel/blob/master/components/camel-test/src/main/java/org/apache/camel/test/junit4/CamelTestSupport.java).
+
+As far as you progress with the development of your test suit you will be able to discover all the features offered by these 2 classes. For the first iteration you will have to just provide the implementation of an abstract method that would take care of building the final route (involving your connector too!). Here is a simple test for the MongoDB:
```
Comment (Delawen) [inline content/docs/connectors.md]: "The more you progress"... better? "As far as" sounds to me like a false friend.

**Comment by Delawen** [comment]
Good work! I think it covers all the basics. They can ask on StackOverflow for more :)

File: content/docs/connectors.md (hunk 2 above)
Comment (squakez) [reply content/docs/connectors.md]: Yes, that's correct. I tend to use commas when it's a kind of list of things, here the meaning is to provide a list of actions (fork, checkout, familiarize)

File: content/docs/connectors.md (hunk 3 above)
Comment (squakez) [reply content/docs/connectors.md]: No, I meant should, as probably there will be some component for the connector that has to be developed.

File: content/docs/connectors.md (hunk 6 above)
Comment (squakez) [reply content/docs/connectors.md]: Probably "as long as" would be better.

**Pull request merged**